### PR TITLE
Remove old DI container

### PR DESCRIPTION
**Pull request by MauricioFauth**
### Description

Removes custom dependency injection container in favor of Symfony Dependency Injection Container.

Related to #15279 and #15266

### CODE REVIEW

**Comment by mauriciofauth** [comment]
@Tithugues Can you review this pull request?

I'm trying to help with the removal of the old DI container.

**Comment by Tithugues** [comment]
@mauriciofauth , thank you! You did almost everything I did yesterday (but didn't push…) so I'm mostly OK with all you that you did. And you also updated the unit tests that wasn't yet up to date on my side!
I believe that I just have a few returns to do on some changes. I'll try to do them today!

File: db_sql.php [hunk 1]
```diff
@@ -19,9 +18,6 @@
```
Comment (Tithugues) [inline db_sql.php]: As this may be an imported file, we may need to add:
```php
global $containerBuilder;
```

File: services.yml [hunk 2]
```diff
@@ -54,6 +57,10 @@ services:
   import:
     class: 'PhpMyAdmin\Import'
 
+  insert_edit:
```
Comment (Tithugues) [inline services.yml]: Thank you for this one! I didn't do it yet on my side!

File: sql.php [hunk 3]
```diff
@@ -27,16 +26,14 @@
```
Comment (Tithugues) [inline sql.php]: As this may be an imported file, we may need to add:
```php
global $containerBuilder;
```

File: tbl_change.php [hunk 4]
```diff
@@ -24,14 +23,11 @@
```
Comment (Tithugues) [inline tbl_change.php]: As this may be an imported file, we may need to add:
```php
global $containerBuilder;
```

File: tbl_export.php [hunk 5]
```diff
@@ -22,14 +21,11 @@
```
Comment (Tithugues) [inline tbl_export.php]: As this may be an imported file, we may need to add:
```php
global $containerBuilder;
```

File: tbl_replace.php [hunk 6]
```diff
@@ -34,14 +33,11 @@
```
Comment (Tithugues) [inline tbl_replace.php]: As this may be an imported file, we may need to add:
```php
global $containerBuilder;
```

File: tbl_select.php [hunk 7]
```diff
@@ -24,14 +22,8 @@
 require_once ROOT_PATH . 'libraries/common.inc.php';
```
Comment (Tithugues) [inline tbl_select.php]: As this may be an imported file, we may need to add:
```php
global $containerBuilder;
```

File: tbl_sql.php [hunk 8]
```diff
@@ -9,20 +9,15 @@
...
 use PhpMyAdmin\Controllers\Table\SqlController;
 use PhpMyAdmin\DatabaseInterface;
-use PhpMyAdmin\Di\Container;
 use PhpMyAdmin\Response;
 use PhpMyAdmin\SqlQueryForm;
-use Symfony\Component\DependencyInjection\Definition;
 
 if (! defined('ROOT_PATH')) {
     define('ROOT_PATH', __DIR__ . DIRECTORY_SEPARATOR);
 }
```
Comment (Tithugues) [inline tbl_sql.php]: As this may be an imported file, we may need to add:
```php
global $containerBuilder;
```

**Review by Tithugues — CHANGES_REQUESTED** [review]

**Comment by Tithugues** [comment]
I think that there is one last think that we can remove… All the classes in `libraries\Di` (with associated tests), except the `Migration` class which is still needed for now.

**Comment by mauriciofauth** [comment]
Thank you @Tithugues for your work and help with replacing the custom DI container.

**Comment by Tithugues** [comment]
Thank you for your help and your reviews.
Feel free to contact me if you find any issues with the new DI. I'll try to help as much as possible.

**Pull request merged**